Mathlib/Algebra/Category/ModuleCat/Presheaf/Sheafify.lean

```diff
@@ -314,10 +314,14 @@ def toSheafify : M₀ ⟶ (restrictScalars α).obj (sheafify α φ).val :=
     simpa using (Sheafify.map_smul_eq α φ (α.app _ r₀) (φ.app _ m₀) (𝟙 _)
       r₀ (by aesop) m₀ (by simp)).symm)
 
-@[simp]
 lemma toSheafify_app_apply (X : Cᵒᵖ) (x : M₀.obj X) :
-    DFunLike.coe (α := M₀.obj X) (β := fun _ ↦ A.val.obj X)
-      ((toSheafify α φ).app X) x = φ.app X x := rfl
+    ((toSheafify α φ).app X).hom x = φ.app X x := rfl
+
+/-- `@[simp]`-normal form of `toSheafify_app_apply`. -/
+@[simp]
+lemma toSheafify_app_apply' (X : Cᵒᵖ) (x : M₀.obj X) :
+    DFunLike.coe (F := (_ →ₗ[_] ↑((ModuleCat.restrictScalars (α.app X)).obj _)))
+    ((toSheafify α φ).app X).hom x = φ.app X x := rfl
```

Comment:
What's going on here?

Reply:
Ah I suppose this is related to the def-eq abuse you mention in the PR description concerning `ModuleCat.restrictScalars` right?

Reply:
It's a bit more complicated but basically yes: we have that the codomain of `((toSheafify α φ).app X).hom` would by default be inferred as some object which is only definitionally equal to `(ModuleCat.restrictScalars (α.app X)).obj _))`. And these definitional equalities are `dsimp` lemmas so they do get applied before `toSheafify_app_apply` (which therefore would never get applied since it looks like it has the wrong type).

One idea I just had: maybe specify `no_index` instead of writing out the expected type tediously?

Reply:
Yes `no_index` works here